core/storage/merkle/polkadot_trie_db/polkadot_codec.cpp

```diff
@@ -17,39 +17,40 @@ OUTCOME_CPP_DEFINE_CATEGORY(kagome::storage::merkle, PolkadotCodec::Error, e) {
       return "number of nibbles in key is >= 2**16";
     case E::UNKNOWN_NODE_TYPE:
       return "unknown polkadot node type";
+    case E::INPUT_TOO_SMALL:
+      return "not enough bytes in the input to decode a node";
   }
 
   return "unknown";
 }
 
 namespace kagome::storage::merkle {
 
-  inline uint8_t low4nibbles(uint8_t byte) {
+  inline uint8_t lowNibble(uint8_t byte) {
     return byte & 0xFu;
   }
 
-  inline uint8_t high4nibbles(uint8_t byte) {
+  inline uint8_t highNibble(uint8_t byte) {
     return (byte >> 4u) & 0xFu;
   }
 
   inline uint8_t collectByte(uint8_t low, uint8_t high) {
     // get low4 from nibbles to avoid check: if(low > 0xff) return error
-    return (low4nibbles(high) << 4u) | low4nibbles(low);
+    return (lowNibble(high) << 4u) | lowNibble(low);
   }
 
   inline common::Buffer ushortToBytes(uint16_t b) {
     common::Buffer out(2, 0);
-    out[0] = b & 0xffu;
-    out[1] = (b >> 8u) & 0xffu;
+    out[0] = (b >> 8u) & 0xffu;
+    out[1] = b & 0xffu;
```

Comment:
why did you decide to swap bytes?

Reply:
It was malfunctioning otherwise. Children indices were messed up.